From b555075a9b3cd377c09971468f3758ab73d50b1e Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Wed, 9 Mar 2016 15:02:21 +0200
Subject: [PATCH 046/239] dram: added support for platform dram setup

Added a platform specific function "ble_plat_dram_setup" that
identifies the DRAM setup and calls the dram initialization
driver.

Change-Id: I38811386bd1780e9cfebfa59b1b18109c98f549f
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28147
Reviewed-by: Haim Boot <hayim@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 ble/ble_main.c                                 |  3 ++
 include/plat/marvell/a8k/common/plat_marvell.h |  3 ++
 plat/marvell/a8k/a7040_rz/dram_setup.c         | 42 ++++++++++++++++++++++++++
 3 files changed, 48 insertions(+)
 create mode 100644 plat/marvell/a8k/a7040_rz/dram_setup.c

diff --git a/ble/ble_main.c b/ble/ble_main.c
index ad1edff..eb0d49e 100644
--- a/ble/ble_main.c
+++ b/ble/ble_main.c
@@ -35,6 +35,7 @@
 #include <debug.h>
 #include <console.h>
 #include <platform_def.h>
+#include <plat_marvell.h>
 
 int  __attribute__ ((section(".entry"))) ble_main(int bootrom_flags)
 {
@@ -42,6 +43,8 @@ int  __attribute__ ((section(".entry"))) ble_main(int bootrom_flags)
 
 	NOTICE("Starting binary extension\n");
 
+	ble_plat_dram_setup();
+
 	return 0;
 }
 
diff --git a/include/plat/marvell/a8k/common/plat_marvell.h b/include/plat/marvell/a8k/common/plat_marvell.h
index c050a7c..62c59de 100644
--- a/include/plat/marvell/a8k/common/plat_marvell.h
+++ b/include/plat/marvell/a8k/common/plat_marvell.h
@@ -82,6 +82,9 @@ void marvell_configure_sys_timer(void);
 /* Topology utility function */
 int marvell_check_mpidr(u_register_t mpidr);
 
+/* BLE utility functions */
+int ble_plat_dram_setup(void);
+
 /* BL1 utility functions */
 void marvell_bl1_early_platform_setup(void);
 void marvell_bl1_platform_setup(void);
diff --git a/plat/marvell/a8k/a7040_rz/dram_setup.c b/plat/marvell/a8k/a7040_rz/dram_setup.c
new file mode 100644
index 0000000..481c7dd
--- /dev/null
+++ b/plat/marvell/a8k/a7040_rz/dram_setup.c
@@ -0,0 +1,42 @@
+/*
+* ***************************************************************************
+* Copyright (C) 2016 Marvell International Ltd.
+* ***************************************************************************
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* Redistributions of source code must retain the above copyright notice, this
+* list of conditions and the following disclaimer.
+*
+* Redistributions in binary form must reproduce the above copyright notice,
+* this list of conditions and the following disclaimer in the documentation
+* and/or other materials provided with the distribution.
+*
+* Neither the name of Marvell nor the names of its contributors may be used
+* to endorse or promote products derived from this software without specific
+* prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+* ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+* LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+* OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+***************************************************************************
+*/
+
+#include <arch_helpers.h>
+#include <plat_marvell.h>
+#include <debug.h>
+
+int ble_plat_dram_setup(void)
+{
+	return 0;
+}
-- 
1.9.1

